riff-raff/app/controllers/Restrictions.scala

```diff
@@ -0,0 +1,86 @@
+package controllers
+
+import java.util.UUID
+import java.util.regex.PatternSyntaxException
+
+import com.gu.googleauth.UserIdentity
+import deployment.{RequestSource, UserRequestSource}
+import org.joda.time.DateTime
+import persistence.RestrictionConfigDynamoRepository
+import play.api.data.Form
+import play.api.data.Forms._
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.libs.ws.WSClient
+import play.api.mvc.{Controller, Result}
+import restrictions.{RestrictionChecker, RestrictionConfig, RestrictionForm}
+import utils.Forms.uuid
+
+class Restrictions()(implicit val messagesApi: MessagesApi, val wsClient: WSClient) extends Controller with LoginActions
+  with I18nSupport {
+
+  lazy val restrictionsForm = Form[RestrictionForm](
+    mapping(
+      "id" -> uuid,
+      "projectName" -> nonEmptyText,
+      "stage" -> nonEmptyText,
+      "editingLocked" -> boolean,
+      "whitelist" -> optional(text),
+      "continuousDeployment" -> boolean,
+      "note" -> nonEmptyText
+    )((id, projectName, stage, editingLocked, whitelist, cdPermitted, note) =>
+      RestrictionForm(id, projectName, stage, editingLocked,
+        whitelist.map(_.split('\n').toSeq.filter(_.nonEmpty)).getOrElse(Seq.empty), cdPermitted, note)
+    )(f =>
+      Some((f.id, f.projectName, f.stage, f.editingLocked, Some(f.whitelist.mkString("\n")),
+        f.continuousDeployment,  f.note))
+    ).verifying(
+      "Stage is invalid - should be a valid regular expression or contain no special values",
+      form => try { form.stage.r; true } catch { case e:PatternSyntaxException => false }
```

Comment:
I thought that would catch fatal errors - but actually Try has that covered which I hadn't realised. Nice idea.